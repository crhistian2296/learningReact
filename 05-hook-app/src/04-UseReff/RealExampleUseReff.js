import React, { useState } from "react";
import MultiplesCustomHooks from "../03-Examples/MultiplesCustomHooks";

const RealExampleUseReff = () => {
  const [show, setShow] = useState(false);

  return (
    <div className="m-5">
      <h1>Cotidian use of useReff</h1>
      <hr />
      {show && <MultiplesCustomHooks />}
      <button
        className="btn btn-primary ml-5"
        onClick={() => {
          setShow(() => !show);
        }}
      >
        show/hide
      </button>
    </div>
  );
};

export default RealExampleUseReff;
