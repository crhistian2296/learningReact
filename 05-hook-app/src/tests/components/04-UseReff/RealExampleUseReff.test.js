import { shallow } from 'enzyme';
import RealExampleUseReff from '../../../04-UseReff/RealExampleUseReff';

describe('Pruebas sobre <RealExampleUseReff/>', () => {
  const wrapper = shallow(<RealExampleUseReff />);

  test('Deberia de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultiplesCustomHooks').exists()).toBeFalsy();
  });

  test('Deberia de mostrar el componente <MultipleCustomHooks/> correctamente', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultiplesCustomHooks').exists()).toBeTruthy();
  });
});
