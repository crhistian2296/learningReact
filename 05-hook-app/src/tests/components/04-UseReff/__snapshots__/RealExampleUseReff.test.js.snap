// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <RealExampleUseReff/> Deberia de mostrar el componente correctamente 1`] = `
<div
  className="m-5"
>
  <h1>
    Cotidian use of useReff
  </h1>
  <hr />
  <button
    className="btn btn-primary ml-5"
    onClick={[Function]}
  >
    show/hide
  </button>
</div>
`;
